lib/screens/post_slot_screen.dart

```diff
@@ -0,0 +1,219 @@
+import 'dart:io';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:uuid/uuid.dart';
+
+class PostSlotScreen extends StatefulWidget {
+  @override
+  _PostSlotScreenState createState() => _PostSlotScreenState();
+}
+
+class _PostSlotScreenState extends State<PostSlotScreen> {
+  String uuid = Uuid().v1();
+  Position position;
+  Position selectedPosition;
+  String selectedLocation;
+  String slotImage;
+  List<Marker> markers = [];
+  User user;
+
+  // Controllers
+  TextEditingController titleController = TextEditingController();
+  TextEditingController descriptionController = TextEditingController();
+  TextEditingController dailyController = TextEditingController();
+  TextEditingController hourlyController = TextEditingController();
+
+  initState() {
+    super.initState();
+    user = FirebaseAuth.instance.currentUser;
+    selectedPosition = position;
+    selectedLocation = 'Tap on map to select location...';
+    getPosition();
+  }
+
+  getPosition() async {
+    Position position = await Geolocator()
+        .getCurrentPosition(desiredAccuracy: LocationAccuracy.high);
+    if (position != null) {
+      setState(() {
+        this.position = position;
+      });
+    }
+  }
+
+  setMarker() {
+    setState(() {
+      markers = [];
+      markers.add(
+        Marker(
+          position:
+              LatLng(selectedPosition.latitude, selectedPosition.longitude),
+          markerId: MarkerId('CurrentSelectedLatLng'),
+        ),
+      );
```

Comment:
Is it correct, that you reset the markers first and add a new marker afterwards?
Does this mean, you can only have a single marker?

I guess it's possible to reduce this to a single command:

```suggestion
      markers = [
        Marker(
          position:
              LatLng(selectedPosition.latitude, selectedPosition.longitude),
          markerId: MarkerId('CurrentSelectedLatLng'),
        ),
      ];
```

Reply:
You cant pass a single marker to the maps-widget, but the user should only be able to select one location at the time.
And yes it is possible to reduce it, thanks.